### DISCUSSION TITLE
Using Aliases with Destinations

### DISCUSSION THREAD

**Discussion by JensRavens** [basecamp/kamal#1116] [Q&A] [answered]
I'm currently giving kamal 2.2 a spin and would like to roll it out with multiple stages (via the destinations feature). The config promotes the feature of aliases, so I created an alias for the rails console:

```yml
aliases:
  console: app exec --interactive --reuse "bin/rails console"
```

Now my question: How do I use an alias in a certain destination? Running `kamal console -d staging` executes the command on the default destination (and sending the argument along to it), while running `kamal -d staging console` fails (with `Could not find command "console".`).

**Comment by dhh** [comment]
cc @djmb

**Comment by djmb** [comment, marked as answer]
Hi @JensRavens,

Setting options before that command name doesn't work for any command - I think this is a limit of Thor, rather than Kamal. It does seem to give a different error for aliases though so I'll look into that.

However, I can't reproduce the problem with the wrong destination being used when calling `kamal console -d staging` - could you post the command output?

**Reply by JensRavens** [reply]
I guess I just had a weird case of a heisen-bug. Just tried it again and the correct destination was used this time. Thanks! :)
